docs/architecture/high-performance-scala.md

```diff
@@ -90,9 +100,45 @@ readability. They've been introduced merely to overcome shortcomings in
 certain boxing-behaviors in the scala compiler and understanding their
 behavior isn't necessary to read/understand the GeoTrellis codebase.
 
+# Micro-Optimizations
+------
 
+## Loops ##
 
-# Macros
------
-# Spark
-------
\ No newline at end of file
+A commonly-seen feature throughout the codebase is the use of `cfor`- or `while`-loops where it would seem that an ordinary `for`-loop would be sufficient.
+We avoid using "`for`-loops" in most cases because they are more than just loops in Scala.
```

Comment:
Dunno how to correct to name `for` a sa `loop`; as it is a `for comprehension`, that's why it is slow; http://docs.scala-lang.org/tutorials/FAQ/yield.html